code/registration.py

```diff
@@ -1211,23 +1208,153 @@ def multiplane_motion_correction(datainput: Path, output_dir: Path, debug: bool
+            "\tsmooth_sigma_time range: "
+            f'{args["smooth_sigma_time_min"]} - '
+            f'{args["smooth_sigma_time_max"]}, '
+            f'steps: {args["smooth_sigma_time_steps"]}'
+        )
+
+        # Create linear spaced arrays for the range of smooth
+        # parameters to try.
+        smooth_sigmas = np.linspace(
+            args["smooth_sigma_min"],
+            args["smooth_sigma_max"],
+            args["smooth_sigma_steps"],
+        )
+        smooth_sigma_times = np.linspace(
+            args["smooth_sigma_time_min"],
+            args["smooth_sigma_time_max"],
+            args["smooth_sigma_time_steps"],
+        )
+
+        optimize_result = optimize_motion_parameters(
+            initial_frames=initial_frames,
+            smooth_sigmas=smooth_sigmas,
+            smooth_sigma_times=smooth_sigma_times,
+            suite2p_args=suite2p_args,
+            trim_frames_start=args["trim_frames_start"],
+            trim_frames_end=args["trim_frames_end"],
+            n_batches=args["n_batches"],
+            logger=logger.info,
+        )
+        if args["use_ave_image_as_reference"]:
+            suite2p_args["refImg"] = optimize_result["ave_image"]
+        else:
+            suite2p_args["refImg"] = optimize_result["ref_image"]
+        suite2p_args["smooth_sigma"] = optimize_result["smooth_sigma"]
+        suite2p_args["smooth_sigma_time"] = optimize_result["smooth_sigma_time"]
+    else:
+        # Create the initial reference image and store it in the
+        # suite2p_args dictionary. 8 iterations is the current default
+        # in suite2p.
+        tic = -time()
+        logger.info("Creating custom reference image...")
+        suite2p_args["refImg"] = compute_reference(
+            input_frames=initial_frames,
+            niter=args["max_reference_iterations"],
+            maxregshift=suite2p_args["maxregshift"],
+            smooth_sigma=suite2p_args["smooth_sigma"],
+            smooth_sigma_time=suite2p_args["smooth_sigma_time"],
+        )
+        tic += time()
+        logger.info(f"took {tic}s")
+    return suite2p_args, args
+
+
+def generate_bergamo_movies(fp: Path, session) -> Path:
+    """Generate virtual movies for Bergamo data
+
+    Parameters
+    ----------
+    fp: Path
+        path to h5 file
+    session: dict
+        session metadata
+    Returns
+    -------
+    Path
+        path to reference image
+    """
+    with h5py.File(fp, "r") as f:
+        data = f["data"][:]
+        dtype = data.dtype
+        # take the first bci epoch to save out reference image TODO
+        tiff_stems = json.loads(f["tiff_stem_location"][:][0])
+        bci_epochs = [
+            i
+            for i in session["stimulus_epochs"]
+            if i["stimulus_name"] == "single neuron BCI conditioning"
+        ]
+        bci_epoch_loc = [i["output_parameters"]["tiff_stem"] for i in bci_epochs][0]
+        with h5py.File("../scratch/reference_image.h5", "w") as ref:
```

Comment:
Can't we just create a VDS? Isn't that a movie not a single image?

Reply:
This is done